### ISSUE TITLE
`flutter run` failed due to permission denied

### ISSUE THREAD

**Issue by gcg1** [flutter/flutter#65378]
When I try to run a newly created flutter app the build fails for iOS simulator referencing the error below. I've been struggling with this for a few days now and any help would be much appreciated.

```
 cp: /Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec: Permission denied
               Command PhaseScriptExecution failed with a nonzero exit code
```

## Steps to Reproduce

1. `flutter create my_app`
2. `cd my_app`
3. `flutter run`

**Expected results:** Build succeeds, demo app opens in simulator.

**Actual results:** Build fails, demo app does not open in simulator.

<summary>Logs</summary>

[pubspec.yaml.txt](https://github.com/flutter/flutter/files/5187983/pubspec.yaml.txt)
[flutterrun--verbose.txt](https://github.com/flutter/flutter/files/5187984/flutterrun--verbose.txt)

`flutter run`

```
Launching lib/main.dart on iPhone 11 in debug mode...
 
Running Xcode build...                                                  
                                                   
Xcode build done.                                            3.7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    cp: /Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec: Permission denied
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone 11.
```

`flutter doctor -v`

```
[✓] Flutter (Channel stable, 1.20.3, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.20.3 at /usr/local/Cellar/flutter/sdk
    • Framework revision 216dee60c0 (6 days ago), 2020-09-01 12:24:47 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/george/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.0

[✓] Connected device (1 available)
    • iPhone 11 (mobile) • 034B59B3-8929-4997-B7A1-53E32D065347 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)

! Doctor found issues in 1 category.

```

</details>

**Comment by pedromassangocode** [comment]
Hi @gcg1 
Can you please provide your `flutter run --verbose` and `pubspec.yaml`. If the logs are too long, you may attach them as a `.txt` file.

Thank you

**Comment by gcg1** [comment]
Added `flutter run --verbose` and `pubspec.yaml` under Logs, thank you.

**Comment by pedromassangocode** [comment]
Hi @gcg1 
Based on the logs provided, I could see that you tried to run you Flutter app on a **iPone 11** and that you have issues with that simulator and others, as I could see in logs:

```
\{\
                            "availabilityError" : "runtime profile not found",\
                            "dataPath" :\
                            "\\/Users\\/george\\/Library\\/Developer\\/CoreSimulator\\\
                            /Devices\\/C4C54B27-C4D7-4795-9150-DFCC90DB4949\\/data\
                            ",\
                            "logPath" :\
                            "\\/Users\\/george\\/Library\\/Logs\\/CoreSimulator\\/C4C5\
                            4B27-C4D7-4795-9150-DFCC90DB4949",\
                            "udid" : "C4C54B27-C4D7-4795-9150-DFCC90DB4949",\
                            "isAvailable" : false,\
                            "deviceTypeIdentifier" :\
                            "com.apple.CoreSimulator.SimDeviceType.iPhone-11",\
                            "state" : "Shutdown",\
                            "name" : "iPhone 11"\
                          \},\
```

And then I noticed that Flutter was not able to find  any supported device to run on.

```
[  +59 ms] \cf4 No supported devices connected.\cf2 \
[  +12 ms] "flutter run" took 4,243ms.\
```

Can you try to run on a physical device?

**Comment by gcg1** [comment]
Thanks for taking a look @pedromassangocode 
Unfortunately I don't have a cable to connect a physical device, I will pick one up within the next day or so and report back. 
I was under the impression there is typically no need for a physical device to `flutter run`.

**Comment by erickzanardo** [comment]
@gcg1 You don't need a physical device to run `flutter run`, simulators work fine. I just noticed that you are using xcode 11.7, that has been released early this month. Try opening xcode and running the project by it, xcode sometime messes up with the simulators through updates.

**Comment by gcg1** [comment]
Thanks for your input @erickzanardo. Running from Xcode targeting iOS Simulator I see the same error: 

```
cp: /Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec: Permission denied
Command PhaseScriptExecution failed with a nonzero exit code

```

[Full_Xcode_Output.txt](https://github.com/flutter/flutter/files/5190474/Full_Xcode_Output.txt)

**Comment by erickzanardo** [comment]
I see, I am updating my xcode here on my machine and will make some tests, I will report back in a few hours (xcode update takes a long time :( )

**Comment by erickzanardo** [comment]
I took a long time, but I finally got able to test. And using the steps describe on the issue description, I couldn't reproduce the issue, so at least I can confirm that the issue is not related to the recently released Xcode. Just for information wise, this is my flutter doctor -v

```
[✓] Flutter (Channel stable, 1.20.3, on Mac OS X 10.15.6 19G2021, locale pt-BR)
    • Flutter version 1.20.3 at /Users/erick.zanardo/flutter
    • Framework revision 216dee60c0 (7 days ago), 2020-09-01 12:24:47 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2


[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/erick.zanardo/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/erick.zanardo/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3
```

**Comment by pedromassangocode** [comment]
Hi @gcg1 
This issue may not be related to Flutter itself. Can you please try the answer at https://stackoverflow.com/a/53383349 and see if it works for you?

**Comment by gcg1** [comment]
Thanks very much for checking that @erickzanardo, good to know.

@pedromassangocode, I tried that solution but it did not fix the issue for me.

**Comment by pedromassangocode** [comment]
Hi @gcg1 
Can you please try the solution on https://github.com/flutter/flutter/issues/39507#issuecomment-555715584 and see if it works for you?

**Comment by pedromassangocode** [comment]
@jmagman just to let you know that this happened again. I saw you worked on https://github.com/flutter/flutter/issues/40146 and https://github.com/flutter/flutter/issues/39507

**Comment by gcg1** [comment]
Hi @pedromassangocode, I'm unable to attempt that solution as the file `ios/Pods/Target\ Support\ Files/Pods-Runner/Pods-Runner-frameworks.sh` does not exist.

Not sure if this is relevant, but my newly created Flutter apps do not have an `ios/Pods` folder by default. I needed to run `pod init` to create a Podfile, then `pod update` to create the Pods folder. After doing so, `flutter run` still fails with the same error.

**Comment by gcg1** [comment]
@pedromassangocode I have just been had a chance to test with a physical iOS device — the build still fails with the same error when the app is run from Xcode or via `flutter run`.

**Comment by erickzanardo** [comment]
> Not sure if this is relevant, but my newly created Flutter apps do not have an `ios/Pods` folder by default. I needed to run `pod init` to create a Podfile, then `pod update` to create the Pods folder. After doing so, `flutter run` still fails with the same error.

As far as I know (and I have just tested to verify), your project will only have a Podfile, and then a Pods folder if your project has any plugins on it. On a just created project, that will not be the case so it is normal for the project to not contain the Pods folder.

**Comment by jmagman** [comment]
@erickzanardo is correct, `ios/Pods` will only exist if you have a Flutter plugin.  @gcg1 Please remove that `ios/Podfile` you created with `pod init`, it's not necessary and will confuse the issue.

`cp: /Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec` is copying the `Flutter.framework` and `Flutter.podspec` from the Flutter SDK, and that gets copied whether or not you have plugins in case it's needed later.  It's not really related to CocoaPods at that step, and it's not related to codesigning.
https://github.com/flutter/flutter/blob/ace54425e089eab797e391ab6469f97cf21dd6c3/packages/flutter_tools/bin/xcode_backend.sh#L156

Does `path/to/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec` exists?  What about `/Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec`?

**Comment by widiu7omo** [comment]
Hey, I also have an issue like @gcg1, then I'm trying to find out if there is a Flutter.podspec file in my project flutter.

I found it but in unwriteable permission.
![Screen Shot 2020-09-11 at 08 05 34](https://user-images.githubusercontent.com/25941348/92830424-e14ba180-f407-11ea-94e9-87f9eefd2ddf.png)

then I try to copy it manually from the terminal.
and he's the result. **the permission is denied.**
![Screen Shot 2020-09-11 at 08 06 17](https://user-images.githubusercontent.com/25941348/92830391-d85ad000-f407-11ea-9d33-2db9a7e85c71.png)

So I added the line on line 142 to delete the Flutter.podspec file from my flutter project
![Screen Shot 2020-09-11 at 08 07 11](https://user-images.githubusercontent.com/25941348/92830527-f9bbbc00-f407-11ea-9a79-1163de1bfe1a.png)

And it works. 
I don't know if it will breaks flow or something else.

**Comment by erickzanardo** [comment]
Maybe this error is related to this other issue? https://github.com/flutter/flutter/issues/65567

The errors are not the same, this issue the author is facing a permission denied, and the other issue is an operation not permitted. But both are related to the Flutter.podspec

**Comment by erickzanardo** [comment]
I have noticed that both on this issue and on the issue I tagged, the flutter sdk is installed on the Desktop folder.

If I recall correctly, that folder has permission restrictions for programs to access it. Maybe that is why you are receiving a permission denied?

Maybe you could try moving the flutter sdk to the home folder and see if the problem persists

**Comment by gcg1** [comment]
> Does path/to/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec exists? What about /Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec?

@jmagman thanks for your input. I can confirm that both of those paths exist. I created a fresh project to ensure no pods/podfile confusion. 

> I added `rm -rf -- "${derived_dir}/Flutter.podspec"` on line 142 to delete the Flutter.podspec file from my flutter project

@widiu7omo thanks for sharing this. I tried adding that line to `xcode_backend.sh` as suggested but it does not fix my issue.

> Maybe you could try moving the flutter sdk to the home folder and see if the problem persists

@erickzanardo my Flutter SDK is already located within the home folder (`HOME/Development/flutter`). 

---

**UPDATE**: After reinstalling Flutter and creating a fresh project my error has changed slightly — still "permission denied" but for
`/Users/george/Desktop/projects/myAppAgain/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/Flutter.framework/Flutter` 
instead of 
`/Users/george/Desktop/projects/myApp/ios/Flutter/Flutter.podspec`
as shown below. I suspect this is due to my previous attempts to force read/write permissions to folders and files.

```
Xcode's output:
↳
    building file list ... done
    deleting App.framework/_CodeSignature/CodeResources
    deleting App.framework/_CodeSignature/
    App.framework/
    App.framework/App

    sent 49767 bytes  received 48 bytes  99630.00 bytes/sec
    total size is 34046879  speedup is 683.47
    building file list ... done
    Flutter.framework/
    Flutter.framework/Flutter
    Flutter.framework/Info.plist
    Flutter.framework/icudtl.dat
    Flutter.framework/_CodeSignature/
    Flutter.framework/_CodeSignature/CodeResources

    sent 98830473 bytes  received 120 bytes  65887062.00 bytes/sec
    total size is 98818019  speedup is 1.00
    /Users/george/Desktop/projects/myAppAgain/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/App.framework/App: signed bundle
    with Mach-O universal (x86_64 arm64) [io.flutter.flutter.app]
    /Users/george/Desktop/projects/myAppAgain/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/Flutter.framework/Flutter:
    replacing existing signature
    /Users/george/Desktop/projects/myAppAgain/build/ios/Debug-iphonesimulator/Runner.app/Frameworks/Flutter.framework/Flutter:
    Permission denied
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
```
[flutter-run--verbose.txt](https://github.com/flutter/flutter/files/5208172/flutter-run--verbose.txt)

**Comment by gcg1** [comment]
Phew, two weeks after running into this issue I was finally been able to resolve it using the steps below:

1. `flutter version v1.11.0`
2. `flutter channel stable` 
3. `flutter upgrade` 
4. `flutter doctor`
5. `flutter create test_app`
6. `cd test_app`
7. `flutter run`

**Comment by mikaelkolkinn** [comment]
@gcg1 Thanks for saving us two weeks of head scratching!

**Comment by jmagman** [comment]
@gcg1 You triggered a re-update to `stable` and the permission problem went away?  

Reopening, I still don't understand what's happening here. Can someone reproducing this include the output of:
```
$ ls -la path/to/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec
$ ls -la path/to/flutter-app/ios/Flutter/Flutter.podspec
```

Mine both shows writable by my user:
```
-rw-r--r--   1 myuser  primarygroup  758 Sep 22 11:18 Flutter.podspec
```
How are you installing Flutter?  Downloading from the [website](https://flutter.dev/docs/get-started/install/macos)?  When I download it from there I still see the cached version writable by my user.

**Comment by gcg1** [comment]
> You triggered a re-update to stable and the permission problem went away?

I listed the steps I took between the build failing with the error and succeeding, beyond that I'm not sure what exactly caused the resolution. Perhaps @mikaelkolkinn can provide some more insight here.

> How are you installing Flutter? Downloading from the website? 

Yes, I downloaded it from the [website](https://flutter.dev/docs/get-started/install/macos). In the process of trying to resolve the issue I tried uninstalling and reinstalling Flutter (by downloading it from the website). 

Here is a list of other things I tried that failed to resolve the issue, in case it's helpful:

* Uninstalling and re-installing Xcode
* Uninstalling and re-installing Flutter
* Running from Xcode
* Removing pods/podfiles and re-installing
* Restarting computer
* `flutter clean`
* Creating a new app
* Checking compliance with all required Flutter "Get started" instructions
* Running from Xcode
* Switching workspace to Legacy Build System
* Locking & unlocking login in Keychain Access
* Giving read/write permissions to the project folder and all enclosed items using Finder, and by running chmod +x on files

**Comment by mikaelkolkinn** [comment]
I cannot provide much more insights other than what @gcg1 has given.
What I can do, is attest that this issue affected our macs, all with fresh installs of Flutter, and that this workaround solved the issue.

**Comment by skyplor** [comment]
I had the exact same issue. but @gcg1's steps helped! So thanks!

Then I tried to change the permissions of Flutter.podspec, giving it +w permission. Then I did `flutter run`. The execution went past the initial erroneous part and I thought it fixed the permission issue, but then immediately after, it gave an error `...build/ios/Debug-iphonesimulator/Runner.app/Frameworks/Flutter.framework/Flutter: Permission denied`

So I tried following @gcg1 to downgrade to v1.11.0 then upgrade back to stable branch (version 1.22.0) and it works!

Here is the exact steps I did to get it working:
$ rm -rf project
$ cd path/to/flutter
$ git co v1.11.0
$ git co stable
$ flutter channel stable
$ cd path/to/project/parent
$ flutter create project
$ flutter run

Something that I found is after I did those steps, the file Flutter.podspec in `path/to/flutter/bin/cache/artifacts/engine/ios/Flutter.podspec` disappeared

**Comment by mike-gee** [comment]
Experienced this issue on v1.23.0

Fixed by switching to v1.22.0
`flutter version 1.22.0`

**Comment by tieorange** [comment]
@mike-gee Same for me today. I can't run my app on macbook 2018. On 2015 works well and I don't have this issue

**Comment by qoli** [comment]
```
[application<com.qoli.uSpace>:64377] This process will be managed, but informing memorystatus failed with error: Operation not permitted (1)
```

```
...
[        ] {"CFBundleName":"uSpace","DTSDKName":"iphonesimulator14.4","DTXcode":"1240","CFBundleIcons~ipad":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60","AppIcon76x76"],"CFBundleIconName":"AppIcon"}},"UILaunchStoryboardName":"LaunchScreen","DTSDKBuild":"18D46","CFBundleDevelopmentRegion":"en","CFBundleVersion":"2","BuildMachineOSBuild":"20D91","DTPlatformName":"iphonesimulator","CFBundlePackageType":"APPL","CFBundleShortVersionString":"1.0.0","CFBundleSupportedPlatforms":["iPhoneSimulator"],"UIMainStoryboardFile":"Main","CFBundleInfoDictionaryVersion":"6.0","CFBundleExecutable":"Runner","DTCompiler":"com.apple.compilers.llvm.clang.1_0","UISupportedInterfaceOrientations~ipad":["UIInterfaceOrientationPortrait","UIInterfaceOrientationPortraitUpsideDown","UIInterfaceOrientationLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"MinimumOSVersion":"12.0","CFBundleIdentifier":"com.qoli.uSpace","UIDeviceFamily":[1,2],"CFBundleSignature":"????","DTPlatformVersion":"14.4","CFBundleIcons":{"CFBundlePrimaryIcon":{"CFBundleIconFiles":["AppIcon60x60"],"CFBundleIconName":"AppIcon"}},"DTXcodeBuild":"12D4e","LSRequiresIPhoneOS":true,"UISupportedInterfaceOrientations":["UIInterfaceOrientationPortrait","UIInterfaceOrientationLandscapeLeft","UIInterfaceOrientationLandscapeRight"],"UIViewControllerBasedStatusBarAppearance":false,"DTPlatformBuild":"18D46"}
[   +5 ms] executing: xcrun simctl launch 6CA57891-E889-45FE-A800-16FDAB32858F com.qoli.uSpace --enable-dart-profiling --enable-checked-mode --verify-entry-points --observatory-port=0
[ +305 ms] com.qoli.uSpace: 60936
[        ] Waiting for observatory port to be available...
```

me too

**Comment by SmitGhori03** [comment]
**Same problem**
error tells to   
Flutter failed to write to a file at "D:\AndroidStudioProjects\tipsy\.flutter-plugins". The flutter tool cannot access the file or directory.
Please ensure that the SDK and/or project is installed in a location that has read/write permissions for the current user.

**Comment by bkonyi** [comment]
Closing as stale. If this is still an issue, please feel free to reopen or file a new issue with details.
